src/graph/planner/plan/Query.h

```diff
@@ -1641,6 +1641,14 @@ class Traverse final : public GetNeighbors {
     firstStepFilter_ = filter;
   }
 
+  Expression* tagFilter() const {
```

Comment:
What's the difference from `vertex filter`?

Reply:
same, but the vertexfilter will not be pushed down to the storage layer, tagFilter will

Reply:
And  vertex filter will be moved to firstStepFilter in the optimizer

Reply:
Too much filter is really confusion.

Reply:
vertexFilter and edgeFilter are in the graph layer. Under some conditions, they cannot be pushed to the storage layer, so they continue to filter at the graph layer. firstStepFilter is the first step version of the filter, which is pushed down to the storage layer, but they cannot filter vertex. So tagFilter is added, if the vertex's attribute does not satisfy tagFilter, it will be filtered out directly in the storage layer

Reply:
We really should not design such a large-grained plan node in the future.